check.sh

```diff
@@ -7,5 +7,10 @@ if [ "${NO_CHANGELOG_LABEL}" = "true" ]; then
 else
     # a changelog check is required
     # fail if the diff is empty
-    ! git diff --exit-code "origin/${BASE_REF}" -- CHANGES.md
+    if git diff --exit-code "origin/${BASE_REF}" -- CHANGES.md; then
```

Comment:
I think the condition here is still wrong, `git diff` will exit with 0 if the files are identical, which by `if` is considered `true` (unlike literally all normal programming languages, but makes sense in shell where exit code 0 is success), so you'll need to negate it.

Reply:
Yes, but isn't this what I want here? If it's zero, than I will print a message and then execute `return 1`

Reply:
I did manual tests before submitting this commit and I'm pretty convinced that it should work. 
This code is simplified to:
```bash
i_return_zero () {
  return 0
}

if i_return_zero ; then
  echo "error"
  exit 1
fi

exit 0
```
and it prints error and returns 1

Reply:
Oh yes, you're completely right - I confused myself by thinking that we want to show the error message when `git diff` reports non-zero, but actually if it is zero then this should be the error and be reported.